experiments/04B.md

```diff
@@ -0,0 +1,234 @@
+# Experiment 4: Segmentation Algorithm Update
+
+## Objective
+
+Update the algorithms used for segmentation to cover the cases that were not covered in the previous experiment (Experiment 3).
+
+---
+
+## Experiment setup
+
+- All the work was done locally without using any accelerators and using Python 3.11 as the kernel of the experiment notebook.
+
+- All the testing and tuning were done on the generated dataset from the Whisper large v3 model with the audio of [sheik AbdulBasit](https://everyayah.com/data/Abdul_Basit_Murattal_192kbps/), like Experiment 3.
+
+- In the middle of this experiment, I found out that I have to see the repeated parts in the reference text of the Holy Quran to cover those cases, so I've done that and the results are at the end of this report.
+
+
+---
+
+## Methodology
+
+### Approach
+
+The approach was to make checkpoints on the reference text using the starting and ending words of each segment of the ayah that the model generated. Then, we would use these checkpoints to segment the reference text into segments that align with the segments generated by the model for each ayah. This can hold for the uncovered cases too but implementing it would be more complicated as we work on the letter level. **So, we will work on the word level.**
+
+---
+
+### code [[1]]
+
+The splitting function:
+
+```python
+def findSplitingIndices_wordwise(aya_segments, ref_aya):

```

Comment:
Stick to snake case, also fix all other occurrences like `findSplitingIndices`, `getAyaSeg`, `segCleaning`, etc.

```suggestion
def find_spliting_indices_wordwise(aya_segments, ref_aya):
```

Reply:
Done